clang/test/3C/tree.c

```diff
@@ -1,6 +1,6 @@
-// RUN: 3c -base-dir=%S -alltypes %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_ALL","CHECK" %s
-// RUN: 3c -base-dir=%S %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_NOALL","CHECK" %s
-// RUN: 3c -base-dir=%S %s -- | %clang -c -fcheckedc-extension -x c -o /dev/null -
+// RUN: 3c --extra-arg="-D_FORTIFY_SOURCE=0" -base-dir=%S -alltypes %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_ALL","CHECK" %s
+// RUN: 3c --extra-arg="-D_FORTIFY_SOURCE=0" -base-dir=%S %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_NOALL","CHECK" %s
+// RUN: 3c --extra-arg="-D_FORTIFY_SOURCE=0" -base-dir=%S %s -- | %clang -c -fcheckedc-extension -x c -o /dev/null -
```

Comment:
When we are not using a compilation database, our convention for the regression tests is to pass compiler options after the `--` rather than via `-extra-arg`:
```suggestion
// RUN: 3c -base-dir=%S -alltypes %s -- -D_FORTIFY_SOURCE=0 | FileCheck -match-full-lines -check-prefixes="CHECK_ALL","CHECK" %s
// RUN: 3c -base-dir=%S %s -- -D_FORTIFY_SOURCE=0 | FileCheck -match-full-lines -check-prefixes="CHECK_NOALL","CHECK" %s
// RUN: 3c -base-dir=%S %s -- -D_FORTIFY_SOURCE=0 | %clang -c -fcheckedc-extension -x c -o /dev/null -
```
Likewise in `type_params.c`.  When I mentioned `-extra-arg` in the meeting, I was thinking of it as a general solution to implement in the `3c` tool for when we don't know whether a compilation database is being used.